src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1179,48 +1197,37 @@ def _to_snowpark_type(self, schema: Tuple[tuple]) -> StructType:
             fields.append(field)
         return StructType(fields)
 
-    def parallel_ingestion(
-        self,
-        create_connection: Callable[[], "Connection"],
-        table: str,
-        column: Optional[str] = None,
-        lower_bound: Optional[Union[str, int]] = None,
-        upper_bound: Optional[Union[str, int]] = None,
-        num_partitions: Optional[int] = None,
-        predicates: Optional[List[str]] = None,
-        *,
-        snowflake_table_type: str = "temporary",
-        snowflake_table_name: Optional[str] = None,
-        use_stored_procedure: bool = False,
-    ) -> DataFrame:
-        pass
-
     def upload_and_copy_into_table(
         self,
         local_file: str,
-        snowflake_stage: str,
-        snowflake_table_type: str = "temporary",
         snowflake_table_name: Optional[str] = None,
-    ):
+    ) -> Optional[Exception]:
+        snowflake_stage = self._session.get_session_stage()
         file_name = os.path.basename(local_file)
-        put_query = f"put file:///Users/yuwang/Desktop/working_repo/snowpark-python/{local_file} @TEST_STAGE/ OVERWRITE=TRUE"
-        copy_into_table_query = f"copy into {snowflake_table_name} from @{snowflake_stage}/{file_name} file_format=(type=parquet) MATCH_BY_COLUMN_NAME=CASE_INSENSITIVE"
-        self._session.sql(put_query).collect()
-        self._session.sql(copy_into_table_query).collect()
+        put_query = f"put file://{local_file} {snowflake_stage}/ OVERWRITE=TRUE"
+        copy_into_table_query = f"copy into {snowflake_table_name} from {snowflake_stage}/{file_name} file_format=(type=parquet) MATCH_BY_COLUMN_NAME=CASE_INSENSITIVE"
+        try:
+            self._session.sql(put_query).collect()
+            self._session.sql(copy_into_table_query).collect()
+            return None
+        except Exception as e:
+            return e
```

Comment:
we should consider retry in case of failure -- both for cost and performance purposes.
if users have 10 GB data split into 10 partitions, 9 GB succeeds but 1GB failed, then the whole ingestion failed.

you can create a separate JIRA to track this

Reply:
got it, logged a jira